lib/phoenix/endpoint/adapter.ex

```diff
@@ -197,7 +198,7 @@ defmodule Phoenix.Endpoint.Adapter do
       end
 
     scheme = url[:scheme] || scheme
-    host   = url[:host]
+    host   = url[:host] || "localhost"
```

Comment:
This is the line that always set the host: https://github.com/phoenixframework/phoenix/blob/master/lib/phoenix/endpoint/adapter.ex#L124